### PR TITLE
GRANITE-26607 - Add package extraction logic

### PR DESCRIPTION
**Pull request by cschneider**

### CODE REVIEW

File: src/main/java/org/apache/sling/distribution/journal/impl/subscriber/ContentPackageExtractor.java [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+import org.osgi.service.component.annotations.Component;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Hook that can be added to a DistributionPackageBuilder.
+ * Each distribution package is inspected for possible content packages in /etc/packages.
+ * Such content packages are installed via the Packaging service.
+ */
+@Component
```
Comment (tmaret) [inline src/main/java/org/apache/sling/distribution/journal/impl/subscriber/ContentPackageExtractor.java]: Is it meant to be a service ? It seems the `ContentPackageExtractor` is instantiated directly in the `DistributionSubscriber`

File: src/main/java/org/apache/sling/distribution/journal/impl/subscriber/ContentPackageExtractor.java [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+
+    private Packaging packageService;
+    private PackageHandling packageHandling;
+    
+    public ContentPackageExtractor(Packaging packageService, PackageHandling packageHandling) {
+        this.packageService = packageService;
+        this.packageHandling = packageHandling;
+    }
+    
+    public void handle(ResourceResolver resourceResolver, List<String> paths) {
```
Comment (tmaret) [inline src/main/java/org/apache/sling/distribution/journal/impl/subscriber/ContentPackageExtractor.java]: PackageHandling.Extract and PackageHandling.Install are not handled differently.

File: src/main/java/org/apache/sling/distribution/journal/impl/subscriber/ContentPackageExtractor.java (hunk 1 above)
Comment (cschneider) [reply src/main/java/org/apache/sling/distribution/journal/impl/subscriber/ContentPackageExtractor.java]: Good catch.. that was a leftover. Will remove it.

File: src/main/java/org/apache/sling/distribution/journal/impl/subscriber/ContentPackageExtractor.java (hunk 2 above)
Comment (cschneider) [reply src/main/java/org/apache/sling/distribution/journal/impl/subscriber/ContentPackageExtractor.java]: They are handled differently below.

**Pull request merged**